elements/nuxeo-document-comments/nuxeo-document-comment-thread.html

```
@@ -0,0 +1,367 @@
+        this.removeEventListener('edit-comment', this._handleEditEvent);
+        this.removeEventListener('comments-changed', this._handleCommentsChange);
+      },
+
+      focusInput: function() {
+        this.$.replyContainer.focus();
+      },
+
+      _clearReply: function() {
+        this.$.selector.clearSelection();
+        this.set('reply', '');
+      },
+
+      _clearRequest: function() {
+        this.$.commentRequest.data = {};
+        this.$.commentRequest.headers = {};
+        this.$.commentRequest.params = {};
+      },
+
+      _fetchComments: function (loadAll) {
+        this._clearRequest();
+        this.$.commentRequest.path = this._computeResourcePath();
+        this.$.commentRequest.params = {
+          pageSize: loadAll ? 0 : this.pageSize,
+          currentPageIndex: 0,
+        };
+        this.$.commentRequest.headers = {
+          "X-NXfetch.comment": "repliesSummary"
+        };
+        this.$.commentRequest.get()
+          .then(function (response) {
+            /* Reconciliation of local and server comments */
+            var olderComment = this.comments.length > 0 ? this.comments[0] : null;
+            var newComments = response.entries;
+            while(newComments.length > 0 && !!olderComment
+            && (newComments[0].creationDate > olderComment.creationDate || newComments[0].id === olderComment.id)) {
+              newComments.shift();
+            }
+            for(var i = 0; i < response.entries.length; i++) {
+              this.unshift('comments', response.entries[i]);
+            }
+
+            this._setTotal(response.totalSize);
+            this._setAllCommentsLoaded(!!loadAll);
+          }.bind(this))
+          .catch(function (error){
+            if(error.status === 404) {
+              this.fire('notify', {message: this._computeTextLabel(this.i18n, this.level, 'notFound')});
+            }
+            console.error(error);
```

Please rethrow the error, if not 404